### PR TITLE
[public-api] Implement ProjectsService.CreateProject

### PR DESCRIPTION
**Pull request by easyCZ**
## Related Issue(s)

* Depends on https://github.com/gitpod-io/gitpod/pull/14624
* Depends on https://github.com/gitpod-io/gitpod/pull/14645

## How to test

1. Preview, sign in with Github and give perms to Read Repo
2. Obtain an Access Token by running the following in the console, after logging in
	```typescript
	await window._gp.gitpodService.server.generateNewGitpodToken({ type: 1, scopes: ["function:getGitpodTokenScopes",
			"function:createTeam",
	        "function:createProject",
			"function:getTeamMembers",
			"function:getGenericInvite",
			"function:getTeams",
			"function:getTeam",
			"resource:default",]})
	```
3. Setup the token as env variable `BEARER_TOKEN="<token>"` in your terminal
4. Create a team programmatically
	```bash
		curl \
	    --header 'Content-Type: application/json' \
	    --header "Authorization: Bearer $BEARER_TOKEN" \
	    --data '{ "name": "<team_name>" }' \
	    https://api.mp-papi-cr5eede70b3a.preview.gitpod-dev.com/gitpod.experimental.v1.TeamsService/CreateTeam
	```
5. Take note of the Team ID in the response, use it in step 6
6. Create a Project in this team, choose one of your GitHub repos
	```
	curl \
	  --header 'Content-Type: application/json' \
	  --header "Authorization: Bearer $BEARER_TOKEN" \
	  --data '{ "name": "connect-go-prometheus", "slug": "connect-go-prometheus", "cloneUrl": "https://github.com/easyCZ/connect-go-prometheus.git", "teamId": "<team-id>", }' \
	  https://api.mp-papi-cr5eede70b3a.preview.gitpod-dev.com/gitpod.experimental.v1.ProjectsService/CreateProject
	```
7. Use the project from the dashboard

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by easyCZ** [comment]
@laushinka 
Thanks. I've copy-pasted the instructions wrong (sorry about that). The right command is:

```bash
curl \
  --header 'Content-Type: application/json' \
  --header "Authorization: Bearer $BEARER_TOKEN" \
  --data '{ "project": {"name": "testing-prebuild-gh", "slug": "testing-prebuild-gh", "cloneUrl": "https://github.com/laushinka/testing-prebuild-gh.git", "teamId": "ee4c2b7e-e7ff-498b-b576-d3ba92d3d01f"} }' \
  https://api.mp-papi-cr5eede70b3a.preview.gitpod-dev.com/gitpod.experimental.v1.ProjectsService/CreateProject
```

**Comment by easyCZ** [comment]
@laushinka Yeah, this relates to https://gitpod.slack.com/archives/C02EN94AEPL/p1668585656105419

**Comment by easyCZ** [comment]
It's possible you did not do step 1. to grant read perms to Github

> 1. Preview, sign in with Github and give perms to Read Repo

**Comment by laushinka** [comment]
> > 1. Preview, sign in with Github and give perms to Read Repo

INDEED. Sorry, let me try again with a different repo.

**Comment by laushinka** [comment]
> > 1. Preview, sign in with Github and give perms to Read Repo
> 
> INDEED. Sorry, let me try again with a different repo.

Like a charm.

/unhold
